command.go

```diff
@@ -1272,8 +1272,8 @@ func (c *Command) InitDefaultHelpCmd() {
 			Short: "Help about any command",
 			Long: `Help provides help for any command in the application.
 Simply type ` + c.DisplayName() + ` help [path to command] for full details.`,
-			ValidArgsFunction: func(c *Command, args []string, toComplete string) ([]string, ShellCompDirective) {
-				var completions []string
+			ValidArgsFunction: func(c *Command, args []string, toComplete string) ([]Completion, ShellCompDirective) {
+				var completions []Completion
```

Comment:
Looks like this broke compatibility as Go is not smart enough to consider `Completion` being an alias for `string`, so things returning `[]string` don't satisfy the interface 🤔 

- https://github.com/docker/cli/issues/5827

Reply:
Thanks for reporting. I’ll look into it

Reply:
For record, the issue wasn't with Completion being an alias of string

But the addition on CompletionFunc type

https://github.com/spf13/cobra/pull/2231#issuecomment-2661425237